arviz/plots/backends/matplotlib/ecdfplot.py

```diff
@@ -0,0 +1,189 @@
+"""Matplotlib ecdfplot."""
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.stats import uniform, binom
+
+from arviz.plots.plot_utils import _scale_fig_size
+from arviz.plots.backends.matplotlib import backend_kwarg_defaults, create_axes_grid, backend_show
+
+
+def plot_ecdf(
+    values,
+    values2,
+    distribution,
+    difference,
+    pit,
+    confidence_bands,
+    granularity,
+    num_trials,
+    alpha,
+    figsize,
+    ecdf_fill,
+    ax,
+    show,
+    backend_kwargs,
+):
+    """Matplotlib ecdfplot."""
+    if backend_kwargs is None:
+        backend_kwargs = {}
+
+    backend_kwargs = {
+        **backend_kwarg_defaults(),
+        **backend_kwargs,
+    }
+
+    (figsize, _, _, _, _, _) = _scale_fig_size(figsize, None)
+    backend_kwargs.setdefault("figsize", figsize)
+    backend_kwargs["squeeze"] = True
+
+    if ax is None:
+        _, ax = create_axes_grid(1, backend_kwargs=backend_kwargs)
+
+    n = len(values)
+
+    if confidence_bands:
+        if pit:
+            x = np.linspace(1 / granularity, 1, granularity)
+            z = x
+            values = (
+                distribution(values)
+                if distribution
+                else compute_ecdf(values2, values) / len(values2)
+            )
+        else:
+            x = np.linspace(values[0], values[-1], granularity)
+            z = distribution(x) if distribution else compute_ecdf(values2, x) / len(values2)
+
+        gamma = get_gamma(n, z, granularity, num_trials, alpha)
+        lower, higher = get_lims(gamma, n, z)
+
+        x_coord, y_coord = [], []
+        if not difference:
+            for _, x_i in enumerate(x):
+                f_x_i = compute_ecdf(values, x_i) / n
+                x_coord.append(x_i)
+                y_coord.append(f_x_i)
+
+            plt.step(x_coord, y_coord, where="post")
+
+            if ecdf_fill:
+                plt.fill_between(x, lower / n, higher / n, color="C0", alpha=0.2)
+            else:
+                plt.plot(x, lower / n, x, higher / n, color="C0", alpha=0.2)
+        else:
+            for _, x_i in enumerate(x):
```

Comment:
It would be preferable if the logic of the loop that incrementally computes the ECDF could be its own function. Then this complicated expression and the code below could be much simpler.

Reply:
Do you mean the for loop on 74th line? Here I am finding the difference in ecdf and ecdf of uniform distribution

Reply:
Yes, but you can do that by 1. constructing the ECDFs and then 2. computing their difference. And all of the blocks that construct an ECDF by looping could be replaced with a single function call that does that, which would simplify the code. Also, that function is then a target for future efficiency improvements.